### PR TITLE
add extractors

### PR DESCRIPTION
**Pull request by wtmount**

### CODE REVIEW

File: src/test/scala/extractors/ExtractorTest.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  it should "return only domains that contain in.ua" in {
+    assert(Extractor.getDomainsInUa(List("something.in.ua", "something.ua")) == List("something.in.ua"))
+  }
+
+  it should "print only emails in gmail.com domain" in {
+    val output = new ByteArrayOutputStream()
+    Console.withOut(output) {
+      Extractor.printEmailsInGmailDomain(Array("email@notgmail.com", "email@gmail.com"))
+    }
+    assert(output.toString == "email@gmail.com\n")
```
Comment (Nazariii) [inline src/test/scala/extractors/ExtractorTest.scala]: The test will fail on windows machine as line ending is hardcoded to Unix style (`\n`).

File: src/main/scala/extractors/Extractor.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  }
+}
+
+object Email {
+  def apply(localPart: String, domain: String): String = localPart + "@" + domain
+
+  def unapply(email: String): Option[(String, String)] = {
+    val parts = email.split("@")
+    if (parts.length == 2) Some(parts(0), parts(1)) else None
```
Comment (Nazariii) [inline src/main/scala/extractors/Extractor.scala]: Pattern matching will be more readable here:
```suggestion
      email.split("@") match {
          case Array(first, second) => Some(first, second)
          case _ => None
    }
```

**Review by Nazariii — CHANGES_REQUESTED** [review]

File: src/test/scala/extractors/ExtractorTest.scala (hunk 1 above)
Comment (wtmount) [reply src/test/scala/extractors/ExtractorTest.scala]: changed to use os specific line separator

File: src/main/scala/extractors/Extractor.scala (hunk 2 above)
Comment (wtmount) [reply src/main/scala/extractors/Extractor.scala]: changed to pattern matching

**Review by Nazariii — APPROVED** [review]

**Pull request merged**